datadog/resource_datadog_synthetics_test_.go

```diff
@@ -1933,6 +1823,163 @@ func buildAssertions(attr []interface{}) []datadogV1.SyntheticsAssertion {
+			optionsScheduling := datadogV1.SyntheticsTestOptionsScheduling{}
+			scheduling := rawScheduling.([]interface{})[0]
+			if rawTimeframes, ok := scheduling.(map[string]interface{})["timeframes"]; ok {
+				var timeFrames []datadogV1.SyntheticsTestOptionsSchedulingTimeframe
+				for _, tf := range rawTimeframes.(*schema.Set).List() {
+					timeframe := datadogV1.NewSyntheticsTestOptionsSchedulingTimeframe()
+					timeframe.SetDay(int32(tf.(map[string]interface{})["day"].(int)))
+					timeframe.SetFrom(tf.(map[string]interface{})["from"].(string))
+					timeframe.SetTo(tf.(map[string]interface{})["to"].(string))
+					timeFrames = append(timeFrames, *timeframe)
+				}
+				optionsScheduling.SetTimeframes(timeFrames)
+			}
+			if timezone, ok := scheduling.(map[string]interface{})["timezone"]; ok {
+				optionsScheduling.SetTimezone(timezone.(string))
+			}
+			options.SetScheduling(optionsScheduling)
+		}
+
+		if retryRaw, ok := d.GetOk("options_list.0.retry"); ok {
+			optionsRetry := datadogV1.SyntheticsTestOptionsRetry{}
+			retry := retryRaw.([]interface{})[0]
+
+			if count, ok := retry.(map[string]interface{})["count"]; ok {
+				optionsRetry.SetCount(int64(count.(int)))
+			}
+			if interval, ok := retry.(map[string]interface{})["interval"]; ok {
+				optionsRetry.SetInterval(float64(interval.(int)))
+			}
+
+			options.SetRetry(optionsRetry)
+		}
+
+		if monitorOptionsRaw, ok := d.GetOk("options_list.0.monitor_options"); ok {
+			monitorOptions := monitorOptionsRaw.([]interface{})[0]
+			optionsMonitorOptions := datadogV1.SyntheticsTestOptionsMonitorOptions{}
+
+			if renotifyInterval, ok := monitorOptions.(map[string]interface{})["renotify_interval"]; ok {
+				optionsMonitorOptions.SetRenotifyInterval(int64(renotifyInterval.(int)))
+			}
+
+			options.SetMonitorOptions(optionsMonitorOptions)
+		}
+
+		if monitorName, ok := d.GetOk("options_list.0.monitor_name"); ok {
+			options.SetMonitorName(monitorName.(string))
+		}
+
+		if monitorPriority, ok := d.GetOk("options_list.0.monitor_priority"); ok {
+			options.SetMonitorPriority(int32(monitorPriority.(int)))
+		}
+
+		if restricted_roles, ok := d.GetOk("options_list.0.restricted_roles"); ok {
+			roles := []string{}
+			for _, role := range restricted_roles.(*schema.Set).List() {
+				roles = append(roles, role.(string))
+			}
+			options.SetRestrictedRoles(roles)
+		}
+
+		if ciRaw, ok := d.GetOk("options_list.0.ci"); ok {
+			ci := ciRaw.([]interface{})[0]
+			testCiOptions := ci.(map[string]interface{})
+
+			ciOptions := datadogV1.SyntheticsTestCiOptions{}
+			ciOptions.SetExecutionRule(datadogV1.SyntheticsTestExecutionRule(testCiOptions["execution_rule"].(string)))
+
+			options.SetCi(ciOptions)
+		}
+
+		if ignoreServerCertificateError, ok := d.GetOk("options_list.0.ignore_server_certificate_error"); ok {
+			options.SetIgnoreServerCertificateError(ignoreServerCertificateError.(bool))
+		}
+
+		// browser tests specific options
```

Comment:
I chose not to check the test type here and to blindly set any options put in the configuration. Even though some options are specific to some test types, the server will not raise any error if some invalid options are used.